frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,442 @@
+      <span>{installMessage}</span>
+      <InstallPageCRDRequiredMessage namespace={namespace} obj={obj} />
+      <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+        {getCRDFromObject(obj) && (
+          <CreateCRDButton namespace={namespace} className="" obj={obj} disabled />
+        )}
+        <ViewInstalledOperatorsButton
+          namespace={namespace}
+          className={getCRDFromObject(obj) ? '' : 'co-clusterserviceversion__link--align-left'}
+        />
+      </ActionGroup>
+    </>
+  );
+};
+
+const OperatorInstallStatus: React.FC<OperatorInstallPageProps> = (props) => {
+  const { resources, targetNamespace, pkgNameWithVersion } = props;
+
+  let loading = true;
+  let status = '';
+  let installObj: ClusterServiceVersionKind | InstallPlanKind =
+    resources?.clusterServiceVersions?.data;
+  const subscription = resources?.subscription?.data;
+  status = installObj?.status?.phase;
+  if (installObj && status) {
+    loading = false;
+  } else if (subscription) {
+    // There is no ClusterServiceVersion for the package, so look at Subscriptions/InstallPlans
+    status = subscription?.status?.state || null;
+    const installPlanName = subscription?.status?.installplan?.name || '';
+    const installPlan: InstallPlanKind = [...resources.installPlan.data].find(
+      (ip) => ip.metadata.name === installPlanName,
+    );
+    if (installPlan) {
+      installObj = installPlan;
+      loading = false;
+    }
+  }
+
+  const isStatusSucceeded = status === 'Succeeded';
+  const isStatusFailed = status === 'Failed';
+  const isApprovalNeeded =
+    installObj?.spec?.approval === 'Manual' && installObj?.spec?.approved === false;
+
+  const approve = () => {
+    k8sPatch(InstallPlanModel, installObj, [
+      { op: 'replace', path: '/spec/approved', value: true },
+    ]).catch((error) => {
+      const errorMsg = error.message;
+      errorModal({ errorMsg });
+    });
+  };
+
+  let indicator = <Spinner size="lg" />;
+  if (isStatusFailed) {
+    indicator = <RedExclamationCircleIcon size="lg" />;
+  }
+  if (isApprovalNeeded) {
+    indicator = <YellowExclamationTriangleIcon size="lg" />;
+  }
+  if (isStatusSucceeded) {
+    indicator = <GreenCheckCircleIcon size="lg" />;
+  }
+
+  let installMessage = <InstallingMessage namespace={targetNamespace} obj={installObj} />;
+  if (isStatusFailed) {
+    installMessage = (
+      <InstallFailedMessage
+        namespace={targetNamespace}
+        obj={installObj}
+        csvName={pkgNameWithVersion}
+      />
+    );
+  }
+  if (isApprovalNeeded) {
+    installMessage = (
+      <InstallNeedsApprovalMessage
+        namespace={targetNamespace}
+        obj={subscription}
+        approve={approve}
+      />
+    );
+  }
+  if (isStatusSucceeded) {
+    installMessage = (
+      <InstallSucceededMessage
+        namespace={targetNamespace}
+        csvName={pkgNameWithVersion}
+        obj={installObj}
+      />
+    );
+  }
+
+  const getCSVLogo = () => {
```

Comment:
I wouldn't make this a function. This seems like a good place to create a separate component instead.